src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -974,17 +980,36 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
         auto uri{ fmt::format(L"{}terminals?cols={}&rows={}&version=2019-01-01&shell={}", _cloudShellUri, _initialCols, _initialRows, shellType) };
 
         WWH::HttpStringContent content{
-            L"",
+            L"{}",
             WSS::UnicodeEncoding::Utf8,
             // LOAD-BEARING. the API returns "'content-type' should be 'application/json' or 'multipart/form-data'"
             L"application/json"
         };
 
-        const auto terminalResponse = _SendRequestReturningJson(uri, content);
+        const auto terminalResponse = _SendRequestReturningJson(uri, content, WWH::HttpMethod::Post(), Windows::Foundation::Uri(_cloudShellUri));
         _terminalID = terminalResponse.GetNamedString(L"id");
 
+        auto socketUri = terminalResponse.GetNamedString(L"socketUri");
+
+        // we have to do some post-handling to get the proper socket endpoint
+        // at this point, socketUri is of the form: wss://ccon-prod-westus-aci-03.servicebus.windows.net/cc-AAAA-AAAAAAAA//aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+        // we need it to become:                    wss://ccon-prod-westus-aci-03.servicebus.windows.net/$hc/cc-AAAA-AAAAAAAA/terminals/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+        // (the "aaaaaa..." part at the end is just the _terminalID, that we already have stored)
+        std::wstring wSocketUri{ socketUri };
+
+        // get the substring up until the ".net"
+        const auto dotNetPos = wSocketUri.find(L".net") + 4;
+        const auto wSocketUriBody = wSocketUri.substr(0, dotNetPos);
+
+        // get the portion between the ".net" and the "//" (this is the cc-AAAA-AAAAAAAA part)
+        const auto lastDoubleSlashPos = wSocketUri.find_last_of(L"//");
+        const auto wSocketUriMiddle = wSocketUri.substr(dotNetPos, lastDoubleSlashPos - (dotNetPos));
+
+        // piece together the final uri, adding in the "$hc" and "terminals" where needed
+        auto finalSocketUri{ fmt::format(L"{}/$hc{}terminals/{}", wSocketUriBody, wSocketUriMiddle, _terminalID) };

```

Comment:
We could also rewrite this as
```cpp
using namespace std::string_view_literals;

static constexpr std::array replacements{
    std::pair{ L".net/"sv, L".net/$hc/"sv },
    std::pair{ L"//"sv, L"/terminals/"sv },
};

for (const auto& [needle, replacement] : replacements)
{
    const auto pos = wSocketUri.find(needle);
    if (pos != decltype(wSocketUri)::npos)
    {
        wSocketUri.replace(pos, needle.size(), replacement);
    }
}
```

That way we only replace what we find without potentially throwing exceptions. I'm still concerned about the reliance on the domain name (.net), but I think that's a smaller concern.

Reply:
This doesn't work because there are 2 instances of `//`: first at the beginning with `wss://` and the second at the end that we actually have to replace